### ISSUE TITLE
Overriding QT Scale factor no longer working in Flameshot 11.0 resulting in broken hidpi scaling

### ISSUE THREAD

**Issue by CocaineJohnsson** [flameshot-org/flameshot#2475]
### Flameshot Version

Flameshot v11.0.0 (-)
Compiled with Qt 5.15.2

### Installation Type

Linux, MacOS, or Windows Package manager (apt, pacman, eopkg, choco, brew, ...)

### Operating System type and version

Arch Linux (kernel 5.16.11) with bspwm

### Description

As of Flameshot 11.0 the workaround found in #1947 no longer appears to work, neither as an export in `.profile` nor as an explicit `QT_SCALE_FACTOR=1.0 flameshot`

`.profile`:
```
XDG_CONFIG_HOME="$HOME/.config"
export XDG_CONFIG_HOME

export QT_AUTO_SCREEN_SCALE_FACTOR=0
export QT_AUTO_SCREEN_SET_FACTOR=0
export QT_SCALE_FACTOR=1.12
export QT_FONT_DPI=92
```

flameshot launched with `QT_SCALE_FACTOR=1.0 flameshot 2>&1 </dev/null &` from bspwmrc

This also results in incorrect UI scaling in the configuration tool
![Screenshot_2022-03-08_13-42-35](https://user-images.githubusercontent.com/53689060/157240403-756b0461-f15d-4e94-974e-07f6a7764fb7.png)

### Steps to reproduce

1. 4K multi-monitor
2. run QT_SCALE_FACTOR=1.0 flameshot gui
3.  you should have an offset

### Screenshots or screen recordings

![20211007_102718](https://user-images.githubusercontent.com/53689060/136348213-6930370b-0fe2-4869-97b2-9f4a80d699ad.jpg)
![20211007_102711](https://user-images.githubusercontent.com/53689060/136348234-7dedf310-9cc0-4ab3-8242-6c3313752b21.jpg)

### System Information

Arch Linux (kernel 5.16.11) with bspwm

![Screenshot_2022-03-08_13-45-42](https://user-images.githubusercontent.com/53689060/157240761-b1940140-264e-4055-a59f-10600ec5e7fd.png)

Bspwm tried with compton on and off, running on xorg.

**Comment by CocaineJohnsson** [comment]
I'm going to leave an update here, I had to reinstall my system because the ext4 driver decided to thrash the entire inode table, 

fractional scaling on QT seems to be an issue, but with a fresh install the old workaround works again (some lingering issue with years of incremental QT updates?) however for whatever reason a fractional scaling of 1.12 (which makes the GUI controls render correctly) worked fine until I installed xorg-xrdb (which I need to configure urxvt to my liking), at which point it exports a DPI value (I have Xft.dpi at 92 to match all the other DPI settings on my system) this cause the behaviour to return, this is likely a QT issue still but is there any way to bypass QT scaling for the capture mode but not for the config window?

Either way, I'm closing this since it seems to have been some issue with my local QT installation and not actually a breakage in flameshot 11.0

**Comment by CocaineJohnsson** [comment]
Well, it actually isn't working when started from bspwmrc with `QT_SCALE_FACTOR=1.0 flameshot 2>&1 </dev/null &`, it works from terminal now though. I'd assume this is a race condition between .profile and .bspwmrc though so I'll just kludge in a little wait before starting bspwmrc, 200ms should be enough.

**Comment by mmahmoudian** [comment]
@CocaineJohnsson thanks for the descriptive updates. Please update us if you got more info.

**Comment by CocaineJohnsson** [comment]
Okay, I did figure something out from further experimentation, if I add `QT_SCALE_FACTOR=1.0` to the hotkey in sxhkdrc:

```
# Screenshot Area
super + shift + s
    QT_SCALE_FACTOR=1.0 flameshot gui
```

it appears to work, even if flameshot is started without an overriden QT_SCALE_FACTOR, since this works even if `flameshot config` is open (with the 1.12 scale factor) that implies the gui is run as a child process?

Did the daemon setting change in 11.0? Flameshot used to daemonise, though if this spawns a new process now that might explain the difference in behaviour. (And indeed, after launching a few instances of flameshot gui I managed to find one in htop's process list before it updated, so it does seem to launch a new process -- I don't know if this was the old behaviour as well, but Qt scaling seems to have been inherited to child processes before, and not now causing the behaviour difference)

![Screenshot_2022-05-26_14-47-48](https://user-images.githubusercontent.com/53689060/170490689-8847dffe-65a6-483d-aaf1-5839b4e31433.png)


Honestly I don't know why xrdb caused it, some interaction with Qt probably but it's likely a red herring from flameshot's perspective.

I suppose the best solution would be if the capture gui would either ignore Qt scaling or handle fractional scaling, the latter to my understanding would be difficult or impossible due to being a Qt issue.


So a rundown:

1) Qt fractional scaling is still partially / fully broken, is it possible to work around?
2) Some difference between 10.x and 11.0 causes child processes to ignore the parent's QT_SCALE_FACTOR (this is both good and bad, since on the one hand it lets the config UI work as intended but means modifying existing hotkeys)
3) xrdb is probably a red herring because that interaction is almost certainly internal to Qt but it had an effect on my system as I was reinstalling (reinstall, pull all the dotfiles, then start installing necessary softwares -- since flameshot was working as intended with my existing unmodified dotfiles until xrdb was installed and the system was rebooted we can infer that xrdb has some interaction with Qt)
